internal/internal_workflow_client.go

```diff
@@ -1792,31 +1779,69 @@ func (w *workflowClientInterceptor) UpdateWorkflow(
 	if err != nil {
 		return nil, err
 	}
-	grpcCtx, cancel := newGRPCContext(ctx, grpcTimeout(pollUpdateTimeout), grpcLongPoll(true), defaultGrpcRetryParameters(ctx))
-	defer cancel()
-	wfexec := &commonpb.WorkflowExecution{
-		WorkflowId: in.WorkflowID,
-		RunId:      in.RunID,
-	}
-	resp, err := w.client.workflowService.UpdateWorkflowExecution(grpcCtx, &workflowservice.UpdateWorkflowExecutionRequest{
-		WaitPolicy:          in.WaitPolicy,
-		Namespace:           w.client.namespace,
-		WorkflowExecution:   wfexec,
-		FirstExecutionRunId: in.FirstExecutionRunID,
-		Request: &updatepb.Request{
-			Meta: &updatepb.Meta{
-				UpdateId: in.UpdateID,
-				Identity: w.client.identity,
-			},
-			Input: &updatepb.Input{
-				Header: header,
-				Name:   in.UpdateName,
-				Args:   argPayloads,
-			},
-		},
-	})
-	if err != nil {
-		return nil, err
+	desiredLifecycleStage := updateLifeCycleStageToProto(in.WaitForStage)
+	var resp *workflowservice.UpdateWorkflowExecutionResponse
+	for ctx.Err() == nil {
```

Comment:
```suggestion
	for {
```

Can probably just let the gRPC call fail as it would with other calls when a context is closed